apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/nfts/[tokenId]/page.tsx

```diff
@@ -0,0 +1,42 @@
+import { notFound, redirect } from "next/navigation";
+import { TokenIdPage } from "../../../../../../../contract-ui/tabs/nfts/components/token-id";
+import { getContractPageParamsInfo } from "../../_utils/getContractFromParams";
+import { getContractPageMetadata } from "../../_utils/getContractPageMetadata";
+
+export default async function Page(props: {
+  params: {
+    contractAddress: string;
+    chain_id: string;
+    tokenId: string;
+  };
+}) {
+  const info = await getContractPageParamsInfo(props.params);
+
+  if (!info) {
+    notFound();
+  }
+
+  const { contract } = info;
+  const { supportedERCs } = await getContractPageMetadata(contract);
+
+  if (!supportedERCs.isERC721 && !supportedERCs.isERC1155) {
+    notFound();
```

Comment:
could probably safely redirect to the `/<chainid>/<contractaddress>` page in this case?

Reply:
updated